pandas/core/groupby.py

```diff
@@ -4280,6 +4280,11 @@ def _nargsort(items, kind='quicksort', ascending=True, na_position='last'):
     if is_categorical_dtype(items):
         return items.argsort(ascending=ascending)
 
+    # # GH11080 - Check monotonic-ness before sort an index
```

Comment:
this optimization is very problematic. Its quite inefficient (we already have these checks on an actual ``Index``, but we don't guarantee this is an index here, though easy to wrap). It doesn't handle nans properly. If it passes the test suite I guess its ok, but would need a performance review. Remove and put in a new PR please.